diff --git a/CMakeLists.txt b/CMakeLists.txt
index 672561e..1ae9c74 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -247,6 +247,8 @@ set_target_properties(snappy
 target_compile_definitions(snappy PRIVATE -DHAVE_CONFIG_H)
 if(BUILD_SHARED_LIBS)
   set_target_properties(snappy PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
+elseif(MSVC)
+  set_target_properties(snappy PROPERTIES OUTPUT_NAME snappy_static)
 endif(BUILD_SHARED_LIBS)
 
 if(SNAPPY_BUILD_TESTS OR SNAPPY_BUILD_BENCHMARKS)
@@ -260,6 +262,10 @@ if(SNAPPY_BUILD_TESTS OR SNAPPY_BUILD_BENCHMARKS)
       "${PROJECT_BINARY_DIR}/config.h"
   )
 
+  if(BUILD_SHARED_LIBS)
+    set_target_properties(snappy_test_support PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
+  endif(BUILD_SHARED_LIBS)
+
   # Test files include snappy-test.h, HAVE_CONFIG_H must be defined.
   target_compile_definitions(snappy_test_support PUBLIC -DHAVE_CONFIG_H)
 
